coalib/results/TextPosition.py

```diff
@@ -7,20 +7,35 @@
 class TextPosition:
 
     @enforce_signature
-    def __init__(self, line: (int, None)=None, column: (int, None)=None):
+    def __init__(self,
+                 line: (int, None)=None,
+                 column: (int, None)=None,
+                 text: (tuple, list, None)=None,
```

Comment:
hmm, so how about we never store text? just have conversion functions? , assert that line, col and position belong to the same character during instantiation?   but then we wont be able to calculate position, line and column, lazily, and the text would be parsed everytime?